### PR TITLE
hubstaff: init at 1.2.14

### PR DESCRIPTION
**Pull request by michalrus**
###### Motivation for this change

Commonly used for time tracking in remote teams.

###### Things done

- [x] Tested using sandboxing
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/hubstaff/default.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+  nativeBuildInputs = [ unzip makeWrapper ];
+
+  unpackCmd = ''
+    # MojoSetups have a ZIP file at the end. ZIP’s magic string is
+    # most often PK\x03\x04. This *should* work for future updates,
+    # but feel free to come up with something more reasonable.
+    dataZipOffset=$(grep --max-count=1 --byte-offset --only-matching --text $'PK\x03\x04' $curSrc | cut -d: -f1)
+    dd bs=$dataZipOffset skip=1 if=$curSrc of=data.zip 2>/dev/null
+    unzip -q data.zip "data/*"
```
Comment (Mic92) [inline pkgs/applications/misc/hubstaff/default.nix]: `7z` could maybe also unpack this without any special arguments.

Comment (michalrus) [reply pkgs/applications/misc/hubstaff/default.nix]: I just tried, but it only unpacks the installer’s files, not the actual data that we need. =(

**Comment by michalrus** [comment]
@Mic92, in the meantime they released 1.2.14.

This has worked well for me for 1½ months now!

**Comment by orivej** [comment]
Files directly in `$out/libexec` are not in place; for example, after installation on NixOS there will be `/run/current-system/sw/libexec/EULA/en.txt`. Could you replace `$out/libexec` with `$out/opt/hubstaff`?

**Comment by michalrus** [comment]
@orivej, absolutely! I’m terribly sorry. I was sure `libexec` was not in `config.environment.pathsToLink`, and that this was it’s whole purpose (not to be linked). Fixed.

**Comment by michalrus** [comment]
@orivej, although, TBH, I can only see this adding `$out/libexec` to linked paths:
https://github.com/NixOS/nixpkgs/blob/a86c8599a464f5718241f79dc52d1cb8ad19b261/nixos/modules/services/mail/opensmtpd.nix#L101

But I didn’t grep multiline.

Are you sure of that?

Just checked, and on my system with this package installed globally (`config.environment.systemPackages = [ hubstaff ];`), `libexec/` is not linked in `/run/current-system/sw/libexec`:
```
% ls /run/current-system/sw 
bin  etc  info  lib  sbin  share
% 
```

**Comment by orivej** [comment]
I have forgotten that it is not linked by default, but I and others configure `environment.pathsToLink = [ "/" ];` and would like to have a neat tree of files in this case too. AFAIK the purpose of the more restricted default is to save some spaces or inodes.

**Comment by michalrus** [comment]
@orivej, thank you very much.

Curiosity: can you elaborate on why having `environment.pathsToLink = [ "/" ];` is useful? Maybe some link? One con I see is that it may prevents you from spotting some bad packaging bugs for some weirdly behaving apps.

**Pull request merged**